### ISSUE TITLE
Rust plugin keeps using "use futures_core::core_reexport::xxx" insted of use "std::"

### ISSUE THREAD

**Issue by crusty-dave** [intellij-rust/intellij-rust#6399]
## Environment

* **IntelliJ Rust plugin version: 0.3.135.3484-202**
* **Rust toolchain version: **
* **IDE name and version: CLion 20.2.4**
* **Operating system: Windows 10**

## Problem description

Rust plugin keeps using "use futures_core::core_reexport::xxx" instead of use "std::" when I use a trait that isn't already included in a module.

i.e. 

`use futures_core::core_reexport::fmt::Debug;`

instead of:

`use std::fmt::Debug;`

## Steps to reproduce

It seems just adding "Default" to a #[cfg(derive(Debug))] triggered the issue.

Cargo.toml contains the following in case it makes a difference:

```toml
[package]
name = "test"
version = "0.1.0"
authors = ["me"]
edition = "2018"

[dependencies]
async-channel = { version = "1.5.1" }
tonic = { version = "0.3", features = ["transport"] }
prost = "0.6"
prost-types = "0.6"
futures = "0.3"
futures-core = "0.3"
futures-util = "0.3"
tokio = { version = "0.2", features = ["macros", "sync", "stream", "time", "rt-core", "rt-util"] }
#icmp = { version = "0.3" }
actix-web = { version = "2.0" }
actix-rt = { version = "1.1" }
actix-http = {version = "1.0" }

chrono = { version = "0.4" }
once_cell = { version = "1.4" }
clap = { version = "2.32" }
json = { version = "0.12" }
serde = { version = "1.0", features = ["derive"] }
serde_derive = { version="1.0" }
serde_json = {  version="1.0" }
log = { version = "0.4" }
log4rs = { version = "0.11" }

async-stream = "0.2"
rand = "0.7"

[build-dependencies]
tonic-build = "0.2"

```

**Comment by Dessix** [comment]
This looks to be the same root problem as is exhibited by #5997, where a private reexport is preferred over the original library (to the point that the original library doesn't occur in autocompletion at all).

**Comment by Dessix** [comment]
Another older occurrence of this I've found is #4000.

**Comment by mattsse** [comment]
> where a private reexport is preferred over the original library (to the point that the original library doesn't occur in autocompletion at all).

prior to 0.3.136 I saw this happening a lot with `async_std`. If I tried to implement `futures::Stream`, i would have typed `impl Stream for X` and then hit autocomplete for the members, however then intellij-rust would auto import `async_std::task::{Task, Context}` and `async_std::pin::Pin` instead auf the actual `std` types. I also saw this happening with `serde::export::Formatter` instead `std::fmt::Fromatter`. It seems that some `std` types that are reexported from crates are being ignored in favor of the reexports crate types. In 0.3.136, I only saw this happening with `async_std::pin::Pin` instead of `std::pin::Pin`
